### ISSUE TITLE
Existing user can't log out and then in with Shibboleth

### ISSUE THREAD

**Issue by acozine** [curationexperts/laevigata#332]
If I log in using Shibboleth, then log out, then log back in again with Shibboleth, I get the "Something Went Wrong" screen. The code is trying to create a new devise user for each login instead of connecting a returning user with the devise record created on her first login. In the production log:
```
D, [2017-06-15T19:20:02.858474 #15965] DEBUG -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2]   SQL (0.7ms)  INSERT INTO "users" ("email", "created_at", "updated_at", "display_name", "provider") VALUES ($1, $2, $3, $4, $5) RETURNING "id"  [["email", "alicia@dce.com"], ["created_at", "2017-06-15 19:20:02.857037"], ["updated_at", "2017-06-15 19:20:02.857037"], ["display_name", "alicia@dce.com"], ["provider", "shibboleth"]]
D, [2017-06-15T19:20:02.858692 #15965] DEBUG -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2]    (0.1ms)  ROLLBACK
I, [2017-06-15T19:20:02.858944 #15965]  INFO -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2] Completed 500 Internal Server Error in 5ms (ActiveRecord: 1.3ms)
F, [2017-06-15T19:20:02.859776 #15965] FATAL -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2]   
F, [2017-06-15T19:20:02.859814 #15965] FATAL -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2] ActiveRecord::RecordNotUnique (PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_users_on_email"
DETAIL:  Key (email)=(alicia@dce.com) already exists.
: INSERT INTO "users" ("email", "created_at", "updated_at", "display_name", "provider") VALUES ($1, $2, $3, $4, $5) RETURNING "id"):
F, [2017-06-15T19:20:02.859845 #15965] FATAL -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2]   
F, [2017-06-15T19:20:02.859870 #15965] FATAL -- : [75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2] app/models/user.rb:27:in `from_omniauth'
[75b7f4c4-7e49-4f01-a068-c5b5f96aeaa2] app/controllers/omniauth_callback_controller.rb:5:in `shibboleth'
```

**Comment by bess** [comment]
I can't duplicate this behavior. I am seeing a different bug... everyone who logs in ends up being Brian, because uid isn't being populated and the "Brian" user exists already with a nil uid. @acozine Can we look at this together?

**Comment by acozine** [comment]
I think both issues were caused by the same thing - a typo I introduced when creating the `clean_shib` branch. I have fixed it (involving some interactive rebasing and squashing) and pushed the fixed code up as `dce_shib`. I have deployed that to stresstest, and I can successfully log in, create a new user with a `uid` attribute, log out, clear the cookies on my browser, and log back in as the `alicia` user. I think this is solved.
